File: typescript/facade/log/console-log.ts
import { Log } from "./log";

export class ConsoleLog implements Log {
    info(message: string): void {
        console.log(`INFO : ${message}`);
    }

    warn(message: string): void {
        console.log(`WARN : ${message}`);
    }

    fatal(message: string): void {
        console.log(`FATAL : ${message}`);
    }
}

File: java/src/main/java/facade/log/ConsoleLog.java
package facade.log;

public class ConsoleLog implements Log {
    @Override
    public void info(String message) {
        System.out.println("INFO : " + message);
    }

    @Override
    public void warn(String message) {
        System.out.println("WARN : " + message);
    }

    @Override
    public void fatal(String message) {
        System.out.println("FATAL : " + message);
    }
}
